[Bug]: npm install Exit handler never called!

### Problem description

I use `npm install` and I get this:

```
npm ERR! Exit handler never called!

npm ERR! This is an error with npm itself. Please report this error at:
npm ERR!     <https://github.com/npm/cli/issues>

npm ERR! A complete log of this run can be found in:
npm ERR!     /data/data/com.termux/files/home/.npm/_logs/2021-12-29T01_28_01_927Z-debug.log
```

when I ignore it and run `npm run serve`, I get this: 

```
node:internal/modules/cjs/loader:936
  throw err;
  ^

Error: Cannot find module '/data/data/com.termux/files/home/Adachi-BOT/...'
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)                                   at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:17:47 {         code: 'MODULE_NOT_FOUND',
  requireStack: []
}
```

### What steps will reproduce the bug?

- run `npm i`
- run `npm run serve`

### What is the expected behavior?

✔ All packages installed (590 packages installed from npm registry, used 23s(network 8s), speed 3.32MB/s, json 516(3.12MB), tarball 23.35MB)

### System information

termux-info:

```
Application version:
0.117
Packages CPU architecture:
aarch64                                                 Subscribed repositories:
# sources.list
deb https://termux.mentality.rip/termux-main/ stable main
# game-repo (sources.list.d/game.list)                  deb https://packages.termux.org/apt/termux-games games stable
# science-repo (sources.list.d/science.list)            deb https://packages.termux.org/apt/termux-science science stable
Updatable packages:                                     All packages up to date
Android version:
11
Kernel build information:
Linux localhost 5.4.86-qgki-g35572e50a678 #1 SMP PREEMPT Wed Oct 27 03:18:25 CST 2021 aarch64 Android
Device manufacturer:
Xiaomi
Device model:
M2102K1AC
```

I use `nodejs-lts`
- node: `v16.13.1`
- npm: `8.1.2`

I try use `nodejs`, but I still get it.
- node: `v17.3.0`
- npm: `8.3.0`

--- a/packages/emscripten/build.sh
+++ b/packages/emscripten/build.sh
@@ -3,7 +3,7 @@ TERMUX_PKG_DESCRIPTION="Emscripten: An LLVM-to-WebAssembly Compiler"
 TERMUX_PKG_LICENSE="MIT"
 TERMUX_PKG_MAINTAINER="@truboxl"
 TERMUX_PKG_VERSION=3.0.1
-TERMUX_PKG_REVISION=1
+TERMUX_PKG_REVISION=2
 TERMUX_PKG_SRCURL=https://github.com/emscripten-core/emscripten.git
 TERMUX_PKG_GIT_BRANCH=$TERMUX_PKG_VERSION
 TERMUX_PKG_PLATFORM_INDEPENDENT=true
@@ -235,12 +235,23 @@ termux_step_make_install() {
 }
 
 termux_step_create_debscripts() {
+	# emscripten's package-lock.json is generated with nodejs v12.13.0
+	# which comes with npm v6 which used lockfile version 1
+	# which isn't compatible with lockfile version 2 used in npm v7 and v8
 	cat <<- EOF > postinst
-	#!$TERMUX_PREFIX/bin/sh
+	#!$TERMUX_PREFIX/bin/bash
 	if [ -n "\$(command -v npm)" ]; then
-	echo 'Running "npm ci --no-optional --production" in $TERMUX_PREFIX/opt/emscripten ...'
 	cd "$TERMUX_PREFIX/opt/emscripten"
+	NPM_VERSION=\$(npm --version)
+	NPM_MAJOR_VERSION=\${NPM_VERSION:0:1}
+	if [ 6 = \$NPM_MAJOR_VERSION ]; then
+	echo 'Running "npm ci --no-optional --production" in $TERMUX_PREFIX/opt/emscripten ...'
 	npm ci --no-optional --production
+	else
+	echo 'Running "npm install" in $TERMUX_PREFIX/opt/emscripten ...'
+	rm package-lock.json
+	npm install
+	fi
 	else
 	echo 'Warning: npm is not installed! Emscripten may not work properly without installing node modules!' >&2
 	fi

--- a/packages/emscripten/build.sh
+++ b/packages/emscripten/build.sh
@@ -248,9 +248,9 @@ termux_step_create_debscripts() {
 	echo 'Running "npm ci --no-optional --production" in $TERMUX_PREFIX/opt/emscripten ...'
 	npm ci --no-optional --production
 	else
-	echo 'Running "npm install" in $TERMUX_PREFIX/opt/emscripten ...'
+	echo 'Running "npm install --no-optional --production" in $TERMUX_PREFIX/opt/emscripten ...'
 	rm package-lock.json
-	npm install
+	npm install --no-optional --production
 	fi
 	else
 	echo 'Warning: npm is not installed! Emscripten may not work properly without installing node modules!' >&2

--- a/packages/nodejs-lts/build.sh
+++ b/packages/nodejs-lts/build.sh
@@ -3,7 +3,7 @@ TERMUX_PKG_DESCRIPTION="Open Source, cross-platform JavaScript runtime environme
 TERMUX_PKG_LICENSE="MIT"
 TERMUX_PKG_MAINTAINER="Yaksh Bariya <yakshbari4@gmail.com>"
 TERMUX_PKG_VERSION=16.13.1
-TERMUX_PKG_REVISION=1
+TERMUX_PKG_REVISION=2
 TERMUX_PKG_SRCURL=https://nodejs.org/dist/v${TERMUX_PKG_VERSION}/node-v${TERMUX_PKG_VERSION}.tar.xz
 TERMUX_PKG_SHA256=4c23004fd75eaf799ad8e76fe34f53e0327f433d4acbfc883396f72e96cc63ad
 # Note that we do not use a shared libuv to avoid an issue with the Android
@@ -86,4 +86,11 @@ termux_step_configure() {
 		$TERMUX_PKG_SRCDIR/out/tools/v8_gypfiles/bytecode_builtins_list_generator.host.mk \
 		$TERMUX_PKG_SRCDIR/out/tools/v8_gypfiles/v8_libbase.host.mk \
 		$TERMUX_PKG_SRCDIR/out/tools/v8_gypfiles/gen-regexp-special-case.host.mk
+}
+
+termux_step_create_debscripts() {
+	cat <<- EOF > ./postinst
+	#!$TERMUX_PREFIX/bin/sh
+	npm config set foreground-scripts true
+	EOF
 }

--- a/packages/nodejs/build.sh
+++ b/packages/nodejs/build.sh
@@ -3,6 +3,7 @@ TERMUX_PKG_DESCRIPTION="Open Source, cross-platform JavaScript runtime environme
 TERMUX_PKG_LICENSE="MIT"
 TERMUX_PKG_MAINTAINER="Yaksh Bariya <yakshbari4@gmail.com>"
 TERMUX_PKG_VERSION=17.3.0
+TERMUX_PKG_REVISION=1
 TERMUX_PKG_SRCURL=https://nodejs.org/dist/v${TERMUX_PKG_VERSION}/node-v${TERMUX_PKG_VERSION}.tar.xz
 TERMUX_PKG_SHA256=e4e4c4e64854698f2590144a177bcc7e7c0befb52020288fdae5c0da0d015d03
 # Note that we do not use a shared libuv to avoid an issue with the Android
@@ -83,4 +84,11 @@ termux_step_configure() {
 		$TERMUX_PKG_SRCDIR/out/tools/v8_gypfiles/bytecode_builtins_list_generator.host.mk \
 		$TERMUX_PKG_SRCDIR/out/tools/v8_gypfiles/v8_libbase.host.mk \
 		$TERMUX_PKG_SRCDIR/out/tools/v8_gypfiles/gen-regexp-special-case.host.mk
+}
+
+termux_step_create_debscripts() {
+	cat <<- EOF > ./postinst
+	#!$TERMUX_PREFIX/bin/sh
+	npm config set foreground-scripts true
+	EOF
 }

